mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoSelectMethodGenerator.java

```diff
@@ -87,32 +93,61 @@ public DaoSelectMethodGenerator(
     if (boundStatementFunction != null) {
       parameters = parameters.subList(0, parameters.size() - 1);
     }
+
+    final List<? extends VariableElement> primaryKeyParameters;
+    final List<? extends VariableElement> freeFormParameters;
     Select selectAnnotation = methodElement.getAnnotation(Select.class);
     assert selectAnnotation != null; // otherwise we wouldn't have gotten into this class
     String customClause = selectAnnotation.customWhereClause();
-    // select without where criteria is ok.
-    // if parameters are provided, we must have at least enough to match partition key.
-    if (customClause.isEmpty()
-        && !parameters.isEmpty()
+    if (parameters.isEmpty()) {
+      primaryKeyParameters = freeFormParameters = Collections.emptyList();
+    } else if (customClause.isEmpty()) {
+      // If we have a partial primary key *and* free-form parameters, things get ambiguous because
+      // we don't know where the primary key ends. By convention, we require all free-form
```

Comment:
>By convention, we require all free-form parameters to be annotated with @CqlName in those cases.

Is it true? from the logic below, it seems that we are treating the first occurrence of the value annotated with `@CqlName` parameter as the limit, and we are not validating that subsequent parameters are annotated with `@CqlName`.

Reply:
Correct. I find that "all non-PK parameters" is almost simpler to explain, but technically it's only the first one. I'll change it.